webapi/CopilotChat/Models/ChatMessage.cs

```diff
@@ -110,6 +110,12 @@ public enum ChatMessageType
     [JsonPropertyName("type")]
     public ChatMessageType Type { get; set; }
 
+    /// <summary>
+    /// Counts of total token usage used to generate bot response.
+    /// </summary>
+    [JsonPropertyName("tokenUsage")]
+    public TokenUsage? TokenUsage { get; set; }
```

Comment:
if we use a default for this then we shouldn't need to pass it in other places
```suggestion
    public TokenUsage? TokenUsage { get; set; } = new TokenUsage(0, 0);
```

Reply:
This was intentional to force token usage calculation to be explicit when creating new bot response message. If it hasn't been explicitly calculated, we show an unknown state since 0 isn't truthy for all scenarios.

Reply:
oh I see, since some responses are hardcoded and are supposed to have 0 token usage vs. sometimes we may fail to calculate it?

Reply:
Could we then have an enum in the TokenUsage class that validates the state of the token usage? i.e. `Calculated, Hardcoded, Unknown`

Reply:
This is not a requirement but for better code readability. Allowing something to be null may need additional code to handle checks and whatnot. And it sounds like null actually means something in this case (unknown token usage), we can better convey that idea by setting an unknown state. this can be done in a separate PR if this PR needs to be merged ASAP.

Reply:
Unknown state is no longer relevant since we the entire ChatAsync method fails if any dependencies fail, so it's not shown in chat history so we don't need to account for it. 

So there's only 2 states, Calculated and Hardcoded, which is already handled with the EmptyTokenUsages() definition.
![image](https://github.com/microsoft/chat-copilot/assets/125500434/973807b8-4ae1-4d40-929e-803647bf6c80)